nemo_aligner/utils/ppo_utils.py

```diff
@@ -87,3 +87,20 @@ def create_mask(values, prompt_lengths, response_lengths):
         # as it is because we want to include one EOS token.
         mask[i, prompt_lengths[i] - 1 : response_lengths[i] - 1] = 1.0
     return mask
+
+
+def select_topk(batch, num_select=1):
+    """
+    Function to select the topk responses for each unique prompt in a batch
+    """
+    unique_prompts = torch.unique(batch["prompt_tokens"], dim=0)
```

Comment:
« O: » This is probably ok for now but can you please add a comment that duplicated prompts in the same batch may cause issues?
(IMO it’d be cleaner to keep track of where the prompts should be, though this would need carrying around more settings)

Reply:
Added